Source/Particles/Collision/BackgroundMCCCollision.cpp

```diff
@@ -0,0 +1,373 @@
+/* Copyright 2021 Roelof Groenewald
+ *
+ * This file is part of WarpX.
+ *
+ * License: ????
+ */
+#include "BackgroundMCCCollision.H"
+#include "MCCScattering.H"
+#include "Utils/ParticleUtils.H"
+#include "Utils/WarpXUtil.H"
+#include "WarpX.H"
+
+// using namespace amrex::literals;
+
+BackgroundMCCCollision::BackgroundMCCCollision (std::string const collision_name)
+    : CollisionBase(collision_name)
+{
+    AMREX_ALWAYS_ASSERT_WITH_MESSAGE(m_species_names.size() == 1,
+               "Background MCC must have exactly one species.");
+
+    amrex::ParmParse pp(collision_name);
+
+    pp.query("background_density", m_background_density);
+    pp.query("background_temperature", m_background_temperature);
+
+    // if the neutral mass is specified use it, but if ionization is
+    // included the mass of the secondary species of that interaction
+    // will be used. If no neutral mass is specified and ionization is not
+    // included the mass of the colliding species will be used
+    m_background_mass = -1;
+    pp.query("background_mass", m_background_mass);
+
+    // query for a list of collision processes
+    // these could be elastic, excitation, charge_exchange, back, etc.
+    amrex::Vector<std::string> scattering_process_names;
+    pp.queryarr("scattering_processes", scattering_process_names);
+
+    // create a vector of MCCProcess objects from each scattering
+    // process name
+    for (auto scattering_process : scattering_process_names) {
+        std::string temp = scattering_process;
+        temp.append("_cross_section");
+        char kw[temp.length() + 1];
+        std::strcpy(kw, temp.c_str());
```

Comment:
This can be written as `auto kw = std::string(scattering_process) + "_cross_section";` and then call `pp.query(kw.c_str(), cross_section_file)`, to fix the MS compiler failure